### PR TITLE
600-799 script commands; bug fixes

### PR DESCRIPTION
**Pull request by Nick007J**

### CODE REVIEW

File: src/peds/Ped.h [hunk 1]
```diff
@@ -339,16 +339,16 @@ class CPed : public CPhysical
 	uint8 bFleeAfterExitingCar : 1;
 
 	uint8 bWanderPathAfterExitingCar : 1;
-	uint8 m_ped_flagF2 : 1;
-	uint8 m_ped_flagF4 : 1; // Unfinished feature from VC, probably bDontDragMeOutCar
+	uint8 bIsLeader : 1;
+	uint8 bDontDragMeOutCar : 1;
```
Comment (erorcun) [inline src/peds/Ped.h]: can you add a comment about that it's unfinished and not working?

File: src/control/Script.cpp [hunk 2]
```diff
@@ -5156,46 +5171,1674 @@ int8 CRunningScript::ProcessCommandsFrom500To599(int32 command)
...
+				continue;
+			if (pPed->GetModelIndex() == MI_SCUM_WOM || pPed->GetModelIndex() == MI_SCUM_MAN)
+				continue;
+			if (!ThisIsAValidRandomPed(pPed->m_nPedType))
+				continue;
+			if (pPed->bIsLeader || pPed->m_leader)
+				continue;
+			if (!pPed->IsWithinArea(x1, y1, x2, y2))
+				continue;
+			if (pos.z - 5.0f > pPed->GetPosition().z)
```
Comment (ShFil119) [inline src/control/Script.cpp]: `5.0f` is used in 4 places, maybe naming it as z offset?

File: src/control/Script.cpp [hunk 3]
```diff
@@ -5156,46 +5171,1674 @@ int8 CRunningScript::ProcessCommandsFrom500To599(int32 command)
...
+	case COMMAND_IS_PLAYER_SHOOTING_IN_AREA:
+	{
+		CollectParameters(&m_nIp, 6);
+		CPed* pPed = CWorld::Players[ScriptParams[0]].m_pPed;
+		assert(pPed);
+		float x1, y1, x2, y2;
+		x1 = *(float*)&ScriptParams[1];
+		y1 = *(float*)&ScriptParams[2];
+		x2 = *(float*)&ScriptParams[3];
+		y2 = *(float*)&ScriptParams[4];
```
Comment (ShFil119) [inline src/control/Script.cpp]:
```
		float x1 = *(float*)&ScriptParams[1];
		float y1 = *(float*)&ScriptParams[2];
		float x2 = *(float*)&ScriptParams[3];
		y2 = *(float*)&ScriptParams[4];
```

File: src/control/Script.cpp [hunk 4]
```diff
@@ -5156,46 +5171,1674 @@ int8 CRunningScript::ProcessCommandsFrom500To599(int32 command)
...
+	case COMMAND_IS_CHAR_SHOOTING_IN_AREA:
+	{
+		CollectParameters(&m_nIp, 6);
+		CPed* pPed = CPools::GetPedPool()->GetAt(ScriptParams[0]);
+		assert(pPed);
+		float x1, y1, x2, y2;
+		x1 = *(float*)&ScriptParams[1];
+		y1 = *(float*)&ScriptParams[2];
+		x2 = *(float*)&ScriptParams[3];
+		y2 = *(float*)&ScriptParams[4];
```
Comment (ShFil119) [inline src/control/Script.cpp]: etc

File: src/control/Script.cpp [hunk 5]
```diff
@@ -5156,46 +5171,1674 @@ int8 CRunningScript::ProcessCommandsFrom500To599(int32 command)
...
+	case COMMAND_GET_RANDOM_CHAR_IN_AREA:
+	{
+		CollectParameters(&m_nIp, 4);
+		int ped_handle = -1;
+		CVector pos = FindPlayerCoors();
+		float x1, y1, x2, y2;
+		x1 = *(float*)&ScriptParams[1];
+		y1 = *(float*)&ScriptParams[2];
+		x2 = *(float*)&ScriptParams[3];
+		y2 = *(float*)&ScriptParams[4];
```
Comment (ShFil119) [inline src/control/Script.cpp]: etc

File: src/control/Script.cpp [hunk 6]
```diff
@@ -6840,6 +7808,34 @@ bool CTheScripts::IsVehicleStopped(CVehicle* pVehicle)
...
+	}
+	if (found) {
+		if (remove)
+			InvisibilitySettingArray[i] = nil;
+	}
+	else if (!remove) {
+		int j = 0;
+		while (i < 20 && !found) {
+			if (InvisibilitySettingArray[i] == nil)
+				found = false;
```
Comment (erorcun) [inline src/control/Script.cpp]: this should be
found = true

**Pull request merged**